marshal_2_tinyint_test.go

```diff
@@ -0,0 +1,81 @@
+package gocql
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/gocql/gocql/marshal/tests/mod"
+	"github.com/gocql/gocql/marshal/tests/serialization"
+	"github.com/gocql/gocql/marshal/tinyint"
+)
+
+func TestMarshalTinyint(t *testing.T) {
+	names := [2]string{
+		"new",
+		"glob",
+	}
+
+	for i, name := range names {
+		t.Run(name, func(t *testing.T) {
+			marshal := tinyint.Marshal
+			unmarshal := tinyint.Unmarshal
+			if i == 1 {
+				marshal = func(i interface{}) ([]byte, error) { return Marshal(NativeType{proto: 4, typ: TypeTinyInt}, i) }
+				unmarshal = func(bytes []byte, i interface{}) error {
+					return Unmarshal(NativeType{proto: 4, typ: TypeTinyInt}, bytes, i)
+				}
+			}
```

Comment:
Do not do it so convoluted:
```suggestion
	tcases := [2]struct{
		name string
		marshal func(interface{}) ([]byte, error)
		unmarshal func(bytes []byte, i interface{}) error
	}{
		{
			name: "serialization.tinyint",
			marshal: tinyint.Marshal,
			unmarshal: tinyint.Unmarshal,
		},
		{
			name: "glob",
			marshal: func(i interface{}) ([]byte, error) {
				return Marshal(NativeType{proto: 4, typ: TypeTinyInt}, i)
			},
			unmarshal: func(bytes []byte, i interface{}) error {
				return Unmarshal(NativeType{proto: 4, typ: TypeTinyInt}, bytes, i)
			},
		},
	}

	for _, tcase := range tcases {
		t.Run(tcase.name, func(t *testing.T) {
```

Reply:
Done